lib/ble/ble.toit

```diff
@@ -0,0 +1,503 @@
+      ble-write-value__ characteristic bytes with-response allow-retry
+
+ble-handle_ resource:
+  #primitive.ble.handle
+
+ble-set-characteristic-notify_ characteristic value:
+  #primitive.ble.set-characteristic-notify
+
+ble-advertise-start_ peripheral-manager name services manufacturer-data interval connection-mode flags:
+  #primitive.ble.advertise-start
+
+ble-advertise-stop_ peripheral-manager:
+  #primitive.ble.advertise-stop
+
+ble-add-service_ peripheral-manager uuid:
+  #primitive.ble.add-service
+
+ble-add-characteristic_ service uuid properties permission value:
+  return ble-run-with-quota-backoff_:
+    ble-add-characteristic__ service uuid properties permission value
+  unreachable
+
+ble-add-characteristic__ service uuid properties permission value:
+  #primitive.ble.add-characteristic:
+    if value == null: throw it
+    return io.primitive-redo-io-data_ it value: | bytes |
+      ble-add-characteristic__ service uuid properties permission bytes
+
+ble-add-descriptor_ characteristic uuid properties permission value:
+  return ble-run-with-quota-backoff_:
+    ble-add-descriptor__ characteristic uuid properties permission value
+
+ble-add-descriptor__ characteristic uuid properties permission value:
+  #primitive.ble.add-descriptor:
+    if value == null: throw it
+    return io.primitive-redo-io-data_ it value: | bytes |
+      ble-add-descriptor__ characteristic uuid properties permission bytes
+
+ble-reserve-services_ peripheral-manager count:
+  #primitive.ble.reserve-services
+
+ble-deploy-service_ service index:
+  #primitive.ble.deploy-service
+
+ble-start-gatt-server_ peripheral-manager:
+  #primitive.ble.start-gatt-server
+
+ble-set-value_ characteristic new-value -> none:
+  ble-run-with-quota-backoff_:
+    ble-set-value__ characteristic new-value
+
+ble-set-value__ characteristic new-value:
+  #primitive.ble.set-value:
+    if new-value == null: throw it
+    return io.primitive-redo-io-data_ it new-value: | bytes |
+      ble-set-value__ characteristic bytes
+
+ble-get-subscribed-clients characteristic:
+  #primitive.ble.get-subscribed-clients
+
+ble-notify-characteristics-value_ characteristic client new-value:
+  return ble-run-with-quota-backoff_:
+    ble-notify-characteristics-value__ characteristic client new-value
+
+ble-notify-characteristics-value__ characteristic client new-value:
+  #primitive.ble.notify-characteristics-value:
+    if new-value == null: throw it
+    return io.primitive-redo-io-data_ it new-value: | bytes |
+      ble-notify-characteristics-value__ characteristic client bytes
+
+ble-get-att-mtu_ resource:
+  #primitive.ble.get-att-mtu
+
+ble-set-preferred-mtu_ adapter mtu:
+  #primitive.ble.set-preferred-mtu
+
+ble-get-error_ characteristic is-oom:
+  #primitive.ble.get-error
+
+ble-clear-error_ characteristic is-oom:
+  #primitive.ble.clear-error
+
+ble-platform-requires-uuid-as-byte-array_:
+  return system.platform == system.PLATFORM-FREERTOS
```

Comment:
Only change: went from `platform` to `system.platform` as I would otherwise need an `import system show platform`, which would then be exported due to the `export *`.